integrations/qdrant/src/haystack_integrations/document_stores/qdrant/converters.py

```diff
@@ -18,23 +21,27 @@ def documents_to_batch(
         documents: List[Document],
         *,
         embedding_field: str,
+        use_sparse_embeddings: bool,
+        sparse_embedding_field: str,
     ) -> List[rest.PointStruct]:
         points = []
         for document in documents:
             payload = document.to_dict(flatten=False)
-            vector = payload.pop(embedding_field) or {}
-            _id = self.convert_id(payload.get("id"))
+            if use_sparse_embeddings:
+                vector = {}
+
+                dense_vector = payload.pop(embedding_field, None)
+                if dense_vector is not None:
+                    vector[DENSE_VECTORS_NAME] = dense_vector
 
-            # TODO: remove as soon as we introduce the support for sparse embeddings in Qdrant
-            if "sparse_embedding" in payload:
-                sparse_embedding = payload.pop("sparse_embedding", None)
-                if sparse_embedding:
-                    logger.warning(
-                        "Document %s has the `sparse_embedding` field set,"
-                        "but storing sparse embeddings in Qdrant is not currently supported."
-                        "The `sparse_embedding` field will be ignored.",
-                        payload["id"],
-                    )
+                sparse_vector = payload.pop(sparse_embedding_field, None)
+                if sparse_vector is not None:
+                    sparse_vector_instance = rest.SparseVector(**sparse_vector)
+                    vector[SPARSE_VECTORS_NAME] = sparse_vector_instance
+
+            if not use_sparse_embeddings:
+                vector = payload.pop(embedding_field) or {}
+            _id = self.convert_id(payload.get("id"))
```

Comment:
Handle previous collection without named dense vector (and without sparse) so that there is not breaking change.